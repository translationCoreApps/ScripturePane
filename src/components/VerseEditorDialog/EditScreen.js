import React from 'react';
import PropTypes from 'prop-types';
import TextField from 'material-ui/TextField';

const styles = {
  input: {
    width: '100%',
    resize: 'none',
    padding: '10px',
    border: 'solid 1px var(--border-color)',
    fontStyle: 'inherit',
    fontVariant: 'inherit',
    fontWeight: 'inherit',
    fontStretch: 'inherit',
    fontSize: 'inherit',
    lineHeight: 'inherit',
    fontFamily: 'inherit',
    cursor: 'inherit',
    outline: 'none',
    backgroundColor: 'transparent',
    WebkitAppearance: 'textfield',
    color: 'rgba(0, 0, 0, 0.870588)'
  }
};

/**
 * Renders a text area for editing the verse
 */
class EditScreen extends React.Component {

  constructor(props) {
    super(props);
    this._handleChange = this._handleChange.bind(this);
  }

  /**
   * Handles changes to the text field
   * @param event
   */
  _handleChange(event) {
    const {onChange} = this.props;
    onChange(event.target.value);
  }

  render() {
    const {verseText} = this.props;
    return (
      <textarea
        id="verse-editor-field"
        rows={4}
        style={styles.input}
        autoFocus={true}
        onChange={this._handleChange}
        value={verseText}/>
    );
  }
}

EditScreen.propTypes = {
  verseText: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
};

export default EditScreen;
