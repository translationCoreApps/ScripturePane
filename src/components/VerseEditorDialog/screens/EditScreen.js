import React from 'react';
import PropTypes from 'prop-types';
import TextField from 'material-ui/TextField';

/**
 * Renders a text area for editing the verse
 */
class EditScreen extends React.Component {

  constructor(props) {
    super(props);
    this._handleChange = this._handleChange.bind(this);
  }

  /**
   * Handles changes to the text field
   * @param event
   */
  _handleChange(event) {
    const {onChange} = this.props;
    onChange(event.target.value);
  }

  render() {
    const {verseText} = this.props;
    return (
      <TextField multiLine={true}
                 rows={4}
                 autoFocus={true}
                 fullWidth={true}
                 onChange={this._handleChange}
                 value={verseText}
                 rowsMax={6}/>
    );
  }
}

EditScreen.propTypes = {
  verseText: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
};

export default EditScreen;
