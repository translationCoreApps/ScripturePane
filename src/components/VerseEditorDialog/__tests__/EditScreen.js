/* eslint-env jest */

import React from 'react';
import renderer from 'react-test-renderer';
import EditScreen from '../EditScreen';

describe('<EditScreen/>', () => {

  it('has text', () => {
    const wrapper = renderer.create(
      <EditScreen onChange={jest.fn()} verseText="verse text"/>
    );
    expect(wrapper).toMatchSnapshot();
  });

});
