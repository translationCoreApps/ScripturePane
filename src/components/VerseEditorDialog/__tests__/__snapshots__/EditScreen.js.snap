// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditScreen/> has text 1`] = `
<textarea
  autoFocus={true}
  id="verse-editor-field"
  onChange={[Function]}
  rows={4}
  style={
    Object {
      "WebkitAppearance": "textfield",
      "backgroundColor": "transparent",
      "border": "solid 1px var(--border-color)",
      "color": "rgba(0, 0, 0, 0.870588)",
      "cursor": "inherit",
      "fontFamily": "inherit",
      "fontSize": "inherit",
      "fontStretch": "inherit",
      "fontStyle": "inherit",
      "fontVariant": "inherit",
      "fontWeight": "inherit",
      "lineHeight": "inherit",
      "outline": "none",
      "padding": "10px",
      "resize": "none",
      "width": "100%",
    }
  }
  value="verse text"
/>
`;
